Analytics/Analytics/Base/UserIdentity.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  UserIdentity.swift
+//  Analytics
+//
+//  Created by Satheesh Kannan on 07/01/25.
+//
+
+import Foundation
+
+/**
+ A struct representing the identity of a user.
+
+ - Properties:
+   - `anonymousId`: A unique identifier for the user when they are not logged in. Defaults to an empty string.
+
+ - Methods:
+   - `initializeState(_:)`: Creates and initializes a `UserIdentity` instance by reading stored data from a key-value storage.
+ */
+public struct UserIdentity {
```

Comment:
It shouldn't be exposed to the user please make it internal!

```suggestion
struct UserIdentity {
```